resolwe_bio/processes/import_data/metadata.py

```diff
@@ -0,0 +1,153 @@
+"""Upload clinical metadata table."""
+from pathlib import Path
+
+import Orange
+
+from resolwe.process import (
+    FileField,
+    IntegerField,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+def format_missing(missing):
+    """Format number of missing values as percentage."""
+    if missing:
+        return f"({missing:.1%} missing values)"
+    else:
+        return "(no missing values)"
+
+
+def get_features_description(data):
+    """Get formated features description."""
+    missing_features = format_missing(
+        data.has_missing_attribute() and data.get_nan_frequency_attribute()
+    )
+    return f"{len(data.domain.attributes)} {missing_features}"
+
+
+def get_target_description(data):
+    """Get formated target class description."""
+    missing_in_class = format_missing(
+        data.has_missing_class() and data.get_nan_frequency_attribute()
+    )
+
+    if data.domain.has_continuous_class:
+        target_description = f"Regression; numerical class {missing_in_class}"
+    elif data.domain.has_discrete_class:
+        target_description = (
+            "Classification; categorical class "
+            f"with {len(data.domain.class_var.values)} values {missing_in_class}"
+        )
+    elif data.domain.class_vars:
+        target_description = (
+            "Multi-target; "
+            f"{len(data.domain.class_vars)} target variables "
+            f"{missing_in_class}"
+        )
+    else:
+        target_description = "Not defined"
+
+    return target_description
+
+
+def change_suffix(path):
+    """Change suffix of a file to lowercase."""
+    new_path = path.with_suffix(path.suffix.lower())
+    path.replace(new_path)
+    return new_path
+
+
+class ImportMetadata(Process):
```

Comment:
I feel `Orange` should be somewhere in the name of the class, process and slug. Maybe `ImportOrangeMetadata`?

Reply:
I agree that this entirely depends on Orange. The name and description have be updated accordingly.